fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java

```diff
@@ -673,8 +673,12 @@ private List<ColumnAccessPath> computeAllColumnAccessPath(PhysicalScanOperator s
                 return scan.getColumnAccessPaths();
             }
 
+            Set<String> checkNames = scan.getColRefToColumnMetaMap().keySet().stream().map(ColumnRefOperator::getName)
+                    .collect(Collectors.toSet());
             if (scan.getPredicate() == null) {
-                return scan.getColumnAccessPaths();
+                // remove path which has pruned
```

Comment:
If the query doesn't have a predicate, will this step remove the path and make the prune field rule not work?

Reply:
It's double check. If a column isn't present in the scanNode's columnMetaMap, it should not be used in query, and set the AccessColumnPath would have no effect.

Reply:
get it